src/modalities/activation_checkpointing.py

```diff
@@ -8,14 +9,18 @@
 )
 from torch.distributed.fsdp.fully_sharded_data_parallel import FullyShardedDataParallel as FSDP
 
-from modalities.models.gpt2.gpt2_model import GPT2Block
 
+import torch
+from typing import List
+
+from modalities.util import get_module_class_from_name
 
-def is_module_to_apply_activation_checkpointing(submodule: torch.nn.Module) -> bool:
-    return isinstance(submodule, GPT2Block)
+def is_module_to_apply_activation_checkpointing(submodule: torch.nn.Module, activation_checkpointing_modules: List[type]) -> bool:
+    return isinstance(submodule, tuple(activation_checkpointing_modules))
 
 
-def apply_activation_checkpointing_inplace(model: torch.nn.Module):
+def apply_activation_checkpointing_inplace(model: torch.nn.Module, activation_checkpointing_modules: List[str]):
+    activation_checkpointing_modules = [get_module_class_from_name(model, m) for m in activation_checkpointing_modules]
     assert isinstance(model, FSDP), "activation checkpointing can only be applied to FSDP wrapped models!"
```

Comment:
```suggestion
    if not isinstance(model, FSDP)
        raise ValueError("activation checkpointing can only be applied to FSDP wrapped models!")
```

asserts normally should not be used in production code, as these checks can be deactivated in optimised code. 

see here: https://stackoverflow.com/questions/56990925/is-this-assert-for-development-not-for-production